app/controllers/announcements_controller.rb

```diff
@@ -3,7 +3,7 @@ class AnnouncementsController < ApplicationController
   before_action :authenticate_user!, except: [:new, :create]
   before_action :set_announcement, only: [:show, :edit, :update, :destroy]
 
-  layout "without_navbar", only: [:new, :create, :show]
+  layout :determine_layout, only: [:new, :create, :show]
```

Comment:
Minor: i _think_ we can drop `:create` from this list since there isn't really a view associated with that action.